homework/assignment_3/spark_connection.py

```diff
@@ -0,0 +1,57 @@
+# libraries
+import sys
+
+from pyspark.sql import SparkSession
+
+
+def main():
+    appName = "PySpark Connection"
+    master = "local"
+
+    # create spark session
+    spark = SparkSession.builder.appName(appName).master(master).getOrCreate()
+
+    # SQL query for 100 day rolling avg
+    sql = """
+        SELECT
+            bc.game_id,
+            SUM(bc.Hit) AS total_hits,
+            SUM(bc.atBat) AS total_atBats,
+            SUM(bc.Hit)/SUM(bc.atBat) AS batting_avg,
+            DATE(g.local_date) AS ora_date,
+            AVG(SUM(bc.Hit)/SUM(bc.atBat)) OVER (ORDER BY g.game_id, DATE(g.local_date)
+            ROWS BETWEEN 100 PRECEDING AND 1 PRECEDING) AS rolling_avg
+        FROM
+            batter_counts bc
+        JOIN
+            game g
+        ON
+            g.game_id = bc.game_id
+        GROUP BY
+            g.game_id"""
+    database = "baseball"
+    user = "root"
+    # pragma: allowlist nextline secret
+    password = "password"  # your password
+    server = "localhost"
+    port = 3306
+    jdbc_url = f"jdbc:mysql://{server}:{port}/{database}?permitMysqlScheme"
+    jdbc_driver = "org.mariadb.jdbc.Driver"
+
+    # create a dataframe by reading data via JDBC
+    df = (
+        spark.read.format("jdbc")
+        .option("url", jdbc_url)
+        .option("query", sql)
+        .option("user", user)
+        # pragma: allowlist nextline secret
+        .option("password", password)
+        .option("driver", jdbc_driver)
+        .load()
+    )
+    # checking fist 15 entries
```

Comment:
You still need to wrap this in a transformer, maybe through using a transformer class in another file and calling it or just using the spark transformer in this file.

Reply:
I think this is also where the calculations may take place because in the assignment page it says that all the logic is done in spark, so finding average here instead of the sql statement maybe.